modules/slugs/includes/class-qtranslate-slug.php

```diff
@@ -353,8 +353,10 @@ function filter_request( $query ) {
         global $q_config;
         global $wp;
 
-        if ( isset( $query['error'] ) && isset( $wp->matched_query ) ) {
-            unset( $query['error'] );
+        if ( isset( $wp->matched_query ) ) {
+            if ( isset( $query['error'] ) ){
+                unset( $query['error'] );
+            }
```

Comment:
I don't understand what we are doing here with `matched_query`. The `error` was not checked in the original plugin. Comments are welcome because it's not obvious.

Reply:
Ref above replies: `error` key was not checked originally because `$query` arg was discarded and replaced completely with `matched_query`, losing all possible additional custom query vars.
In previous commit I tried to limit the use of `matched_query` to the cases where nothing could be found with `$query` only (i.e. `'error' => 404`), but it seems in certain cases (e.g. pages) `matched_query` data is strictly needed.
Hence `matched_query` is now always merged with `$query` removing from the latter specific keys (e.g. `error`);